### PR TITLE
Make C/Fortran compilers optional

### PR DESCRIPTION
**Pull request by havogt**
Introduce `CPP_BINDGEN_REQUIRE_TEST_Fortran` and `CPP_BINDGEN_REQUIRE_TEST_C` to force C and Fortran compilers, otherwise the respective tests are only enabled if  available.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Review by lukasm91 — APPROVED** [review]

File: tests/regression/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-    add_subdirectory(array_gt_legacy)
-endif()
+if(CMAKE_Fortran_COMPILER_LOADED)
+    add_subdirectory(array)
+    if(CPP_BINDGEN_GT_LEGACY)
+        add_subdirectory(array_gt_legacy)
+    endif()
 
-add_subdirectory(simple)
+    add_subdirectory(simple)
```
Comment (lukasm91) [inline tests/regression/CMakeLists.txt]: Is this correct? Shouldn't this be outside the if? (Ther are C-tests)

**Review by lukasm91 — CHANGES_REQUESTED** [review]

File: tests/regression/CMakeLists.txt (hunk 1 above)
Comment (havogt) [reply tests/regression/CMakeLists.txt]: Yes, thanks. Maybe I should move it inside the subfolders for everything, but then it is less visible. So for now I'll fix it but keep the logic for the other folders here.

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: tests/regression/CMakeLists.txt (hunk 1 above)
Comment (lukasm91) [reply tests/regression/CMakeLists.txt]: Looks good. I'll make a note to file a bug report to CMake when I have some free minutes (the issues with check_language)

**Review by lukasm91 — APPROVED** [review]

**Pull request merged**